sw/host/opentitanlib/src/transport/hyperdebug/mod.rs

```diff
@@ -0,0 +1,320 @@
+                                Self::find_tty(&interface_path)?);
+                        } else if interface_name == "SPI" {
+                            // We found the SPI forwarding USB interface (this one interface
+                            // allows multiplexing physical SPI ports.)
+                            let mut in_endpoint: u8 = 0;
+                            let mut out_endpoint: u8 = 0;
+                            for endpoint_desc in interface_desc.endpoint_descriptors() {
+                                if endpoint_desc.transfer_type() == rusb::TransferType::Bulk {
+                                    match endpoint_desc.direction() {
+                                        rusb::Direction::In => {
+                                            in_endpoint = endpoint_desc.address();
+                                        }
+                                        rusb::Direction::Out => {
+                                            out_endpoint = endpoint_desc.address();
+                                        }
+                                    }
+                                }
+                            }
+                            if in_endpoint == 0 || out_endpoint == 0 {
+                                return Err(Error::CommunicationError("Missing SPI endpoint").into());
+                            }
+                            spi_interface.replace(BulkInterface {
+                                interface: interface.number(),
+                                in_endpoint,
+                                out_endpoint,
+                            });
+                        }
+                    }
+                }
+            }
+        }
+        let mut spi_names = HashMap::new();
+        // Eventually, the SPI aliases below should either go into configuration file, or come
+        // from the HyperDebug firmware, declaring what it supports (as is the case with UARTs.)
+        spi_names.insert("SPI2".to_string(), 0);
+        spi_names.insert("0".to_string(), 0);
+        let result = Hyperdebug {
+            spi_names,
+            spi_interface: spi_interface.ok_or(Error::CommunicationError("Missing SPI interface"))?,
+            uart_ttys,
+            inner: Rc::new(Inner {
+                console_tty: console_tty.ok_or(Error::CommunicationError("Missing console interface"))?,
+                usb_device: RefCell::new(device),
+                gpio: Default::default(),
+                spis: Default::default(),
+                uarts: Default::default(),
+            })
+        };
+        Ok(result)
+    }
+
+    /// Locates the /dev/ttyUSBn node corresponding to a given interface in the sys directory
+    /// tree, e.g. /sys/bus/usb/devices/1-4/1-4:1.0 .
+    fn find_tty(path: &Path) -> Result<PathBuf> {
+        for entry in fs::read_dir(path)? {
+            let dir_entry = entry?;
+            match dir_entry.file_name().into_string() {
```

Comment:
This looks like it would be happier as `if let Ok(filename) = ... { ... }`

Reply:
Of course, thanks.